### PR TITLE
Fixed problem with missing metrics in decoder

### PR DESCRIPTION
**Pull request by vishh**
Cherrypick #623 into v0.18 release branch

cc @piosz @mwielgus

### CODE REVIEW

**Comment by piosz** [comment]
LGTM

**Comment by piosz** [comment]
Thanks for doing it!

File: sinks/api/decoder.go [hunk 1]
```diff
@@ -120,9 +125,9 @@ func (self *decoder) getContainerMetrics(container *cache.ContainerElement, labe
 				Name:   supported.Name,
 				Labels: labelsAsString,
 			}
-			// TODO: remove this once the heapster source is tested to not provide duplicate metric.Statss.
 
-			if data, ok := self.lastExported[key]; ok && data.After(metric.Stats.Timestamp) {
+			// TODO: remove this once the heapster source is tested to not provide duplicate metric.Stats.
```
Comment (vishh) [inline sinks/api/decoder.go]: Just noticed this @piosz. The last time I tested, I did not see any duplicate stats. We can add a Vlog here to validate my claim.

Comment (piosz) [inline sinks/api/decoder.go]: I tested this both manually and the unit test below and I haven't seen any duplicates.

**Pull request merged**